modules/audit-enqueued-assets/load.php

```diff
@@ -0,0 +1,270 @@
+ *
+ * @since 1.0.0
+ */
+function perflab_aea_audit_enqueued_styles() {
+	if ( ! is_admin() && ! get_transient( 'aea_enqueued_styles' ) ) {
+		global $wp_styles;
+		$enqueued_styles = array();
+		foreach ( $wp_styles->queue as $handle ) {
+			$src = $wp_styles->registered[ $handle ]->src;
+			if ( $src && ! strpos( $src, 'wp-includes' ) ) {
+				$enqueued_styles[] = $src;
+			}
+		}
+		$expiration = apply_filters( 'perflab_aea_audit_enqueued_styles_expiration_in_seconds', 12 * HOUR_IN_SECONDS );
+		set_transient( 'aea_enqueued_styles', $enqueued_styles, $expiration );
+	}
+}
+add_action( 'wp_print_styles', 'perflab_aea_audit_enqueued_styles' );
+
+/**
+ * Gets total of enqueued scripts.
+ *
+ * @since 1.0.0
+ *
+ * @return int|false Number of total scripts or false if transient hasn't been set.
+ */
+function perflab_aea_get_total_enqueued_scripts() {
+	$enqueued_scripts      = false;
+	$list_enqueued_scripts = get_transient( 'aea_enqueued_scripts' );
+	if ( $list_enqueued_scripts ) {
+		$enqueued_scripts = count( $list_enqueued_scripts );
+	}
+	return $enqueued_scripts;
+}
+
+/**
+ * Gets total of enqueued styles.
+ *
+ * @since 1.0.0
+ *
+ * @return int|false Number of total styles or false if transient hasn't been set.
+ */
+function perflab_aea_get_total_enqueued_styles() {
+	$enqueued_styles      = false;
+	$list_enqueued_styles = get_transient( 'aea_enqueued_styles' );
+	if ( $list_enqueued_styles ) {
+		$enqueued_styles = count( $list_enqueued_styles );
+	}
+	return $enqueued_styles;
+}
+
+/**
+ * Adds tests to site health.
+ *
+ * @since 1.0.0
+ *
+ * @param array $tests Site Health Tests.
+ * @return array
+ */
+function perflab_aea_add_enqueued_assets_test( $tests ) {
+	$tests['direct']['enqueued_js_assets']  = array(
+		'label' => esc_html__( 'JS assets', 'performance-lab' ),
+		'test'  => 'perflab_aea_enqueued_js_assets_test',
+	);
+	$tests['direct']['enqueued_css_assets'] = array(
+		'label' => esc_html__( 'CSS assets', 'performance-lab' ),
+		'test'  => 'perflab_aea_enqueued_css_assets_test',
+	);
+
+	return $tests;
+}
+add_filter( 'site_status_tests', 'perflab_aea_add_enqueued_assets_test' );
+
+/**
+ * Callback for enqueued_js_assets test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function perflab_aea_enqueued_js_assets_test() {
+	/**
+	 * If the test didn't run yet, deactivate.
+	 */
+	$enqueued_scripts = perflab_aea_get_total_enqueued_scripts();
+	if ( false === $enqueued_scripts ) {
+		return array();
+	}
+
+	$result = array(
+		'label'       => esc_html__( 'Enqueued JS assets', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__( 'The amount of enqueued JS assets is acceptable.', 'performance-lab' )
+		),
+		'actions'     => '',
+		'test'        => 'enqueued_js_assets',
+	);
+
+	if ( $enqueued_scripts > 10 ) {
```

Comment:
This signal needs to be more robust. The number of scripts is somehow a week signal, what really matter is the totally size. We need to grab and sum the size of all scripts and define a threshold for the warning.

Reply:
Hello @ThierryA, should we combine both signals? number + size?  I think it stills makes sense to know the number of assets, so we can recommend concatenating them for instance. 

Should we define the size threshold and messaging?

Reply:
@ThierryA I've implemented the logic to calculate sizes and so, we just need to define the thresholds and messaging.

Reply:
Great, let's discuss with the broader group about the thresholds for scripts and stylesheets size as well as the number if we decide to have one.

https://almanac.httparchive.org/en/2021/javascript#how-much-javascript-do-we-load and https://almanac.httparchive.org/en/2021/css can help giving us an understanding of the current state of the web.

We could base the threshold on CWV guidance if that make sense to everyone.

@felixarntz @adamsilverstein paging you here as well, would love more people thoughts on this too.

Reply:
Thanks @manuelRod. Let's have a discussion with the broader group about the scripts and stylesheets size threshold as well as the number of instance if we decide to include this as a signal too.

https://almanac.httparchive.org/en/2021/javascript#how-much-javascript-do-we-load and https://almanac.httparchive.org/en/2021/css may help us understanding the current state of the web

We could base the threshold on CWV guidance if that resonate with everyone.

@felixarntz @adamsilverstein paging you here as well, would love more folks thoughts on this too.

Reply:
+1, we should warn if _either_ the number of scripts _or_ the overall size of the scripts is "too high". But it is indeed a complicated question what our thresholds should be for both of those, let's continue discussing that on the issue.

Maybe for now it would still be good to already change the code here to make the intention clear - maybe we start with a threshold of 100KB, just to have something to begin with and further prepare the code?

Reply:
closing since they are outdated